src/middleware/data-factory.ts

```diff
@@ -0,0 +1,16 @@
+import {
+  DefaultContext, DefaultState, Middleware, Next,
+} from 'koa';
+import { DataFactory } from 'rdf-js';
+
+export type DataFactoryContext<CustomT extends object = DefaultContext,
+  CustomF extends object = {}> = CustomT & { dataFactory: DataFactory & CustomF };
```

Comment:
`T` and `F` don't sound very explanatory, they stand for `Type` and?

Reply:
`CustomT` using same name as Koa. `F` for factory, but these could be better.